Feat: User inputs validation

DevdouglasGfs opened the pull request:
# User Inputs Validation

Don't allow the form submission with invalid user inputs

This solves the issue #7

DevdouglasGfs committed:
Docs: Update README.md file

DevdouglasGfs committed:
Fix: Remove unused section
Remove "Instructions" section and fix the project name

DevdouglasGfs committed:
Feat: Add user input validations
Don't allow submit the form if the user input isn't a valid format

walteraandrade commented on src/components/login-form/index.tsx:
```diff
@@ -0,0 +1,88 @@
+                className='input-group__input'
+              />
+              {showValidationMessage && password.trim().length < 7 && (
+                <p className='input-group__informative-message'>A senha deve ter ao menos 7 caractéres.</p>
```
```suggestion
                <p className='input-group__informative-message'>A senha deve ter ao menos 7 caracteres.</p>
```

walteraandrade commented on src/components/login-form/index.tsx:
```diff
@@ -0,0 +1,88 @@
+                <p className='input-group__informative-message'>A senha deve ter ao menos 7 caractéres.</p>
+              )}
+              {showValidationMessage && !validPassword && (
+                <p className='input-group__informative-message'>A senha deve ser composta por caractéres alfánumericos.</p>
```
```suggestion
                <p className='input-group__informative-message'>A senha deve ser composta por caracteres alfánumericos.</p>
```

walteraandrade commented on src/components/login-form/index.tsx:
```diff
@@ -0,0 +1,88 @@
+              {showValidationMessage && !validPassword && (
+                <p className='input-group__informative-message'>A senha deve ser composta por caractéres alfánumericos.</p>
+              )}
+              {validPassword ? 'true' : 'false'}
```
não entendi isso 🤔

DevdouglasGfs replied:
Ah, quanto a isso foi um teste que fiz para vêr se estava se comportando de acordo 😁 apaguei na outra PR

DevdouglasGfs committed:
Chore: Writting correction

DevdouglasGfs committed:
Fix: use validators to validate user inputs

SamuelSsc reviewed (changes requested):
Esse pr está comparando com a branch errada como base, como consequência aparecem implementações que você fez em outro momento, ao abrir um pr ajusta na parte de cima para onde ele está apontando Ex esse de dev aqui abaixo: 
<img width="405" alt="image" src="https://github.com/DevdouglasGfs/taqtile/assets/60801255/adb1d96e-0f53-4469-bfef-2ffa380c789e">

No seu caso se você criou essa branch logo depois da de login você já abre ela apontando para de login, dessa forma só vai aparecer alterações dessa branch aqui especificamente.

DevdouglasGfs commented:
Ahhh, muito obrigado! Não sabia disso

DevdouglasGfs commented:
Pronto, fix o rebase necessário. Obrigado pelo auxílio novamente 💯

DevdouglasGfs commented:
Posso dar merge?

DevdouglasGfs commented:
Assim que é feita a validação nesta branch
![Screenshot_20240626-122342](https://github.com/DevdouglasGfs/taqtile/assets/105404079/d3b5124b-d74d-4a41-9cba-d5d917aa383f)
![Screenshot_20240626-122639](https://github.com/DevdouglasGfs/taqtile/assets/105404079/0f56506a-519d-4317-b3d6-1aa1b5ef68aa)

DevdouglasGfs committed:
Feat: Add file validators file
The file validators.ts inside of the utils folder contains validators;
The validators include patterns and functions for validations;
